src/tests/e2e/api/oauth2_api_e2e_test.py

```diff
@@ -62,3 +66,29 @@ def test_login_user_successfully(self):
             request_as_json.get("user").get("roles"),
         )
         self.assertEqual(token_decoded.get("scopes"), request_as_json.get("scopes"))
+
+    def test_refresh_token_endpoint_with_valid_access_token(self):
+
+        request_body = create_refresh_token_request(
+            self.engine, secret_key=self.env_config.oauth2_secret_key
+        )
+
+        request = self.client.post("/refresh_token", json=request_body)
+        request_as_json = request.get_json()
+        self.assertEqual(
+            request_body.get("access_token"), request_as_json.get("access_token")
+        )
+
+    def test_refresh_token_endpoint_with_expired_access_token(self):
+
+        expired_access_token = build_expire_access_token()
+        request_body = create_refresh_token_request(
+            self.engine,
+            secret_key=self.env_config.oauth2_secret_key,
+            access_token=expired_access_token,
+        )
+        request = self.client.post("/refresh_token", json=request_body)
+        request_as_json = request.get_json()
+        self.assertNotEqual(
+            request_body.get("access_token"), request_as_json.get("access_token")
```

Comment:
could you confirm the generated token is valid?

Reply:
Done.